pipeline/data_store.py

```diff
@@ -515,11 +515,11 @@ def get_image(self, provenance=None, session=None):
             # (If all the ids are None, it'll match even if the actual
             # objects are wrong, but, oh well.)
             if (self.exposure_id is not None) and (self.section_id is not None):
-                if ( (self.image.exposure_id != self.exposure_id) or
+                if ( (self.image.exposure_id is not None and self.image.exposure_id != self.exposure_id) or
```

Comment:
What's the reason for this change?  If self.image.exposure_id is None, then shouldn't self.exposure_id also be None?

Reply:
There are some situations (e.g., where Image is loaded from the test cache without the proper links) where it would fail before having a chance to assign all the new relationships.